google-home-community.groovy

```diff
@@ -1489,38 +1491,68 @@ private deviceTraitPreferences_SensorState(deviceTrait) {
         deviceTrait.sensorTypes.each { sensorType ->
             // only display if the sensor has descriptive values
             if (GOOGLE_SENSOR_STATES[sensorType.key].descriptiveState) {
-                input(
-                    name: "${deviceTrait.name}.sensorTypes.${sensorType.key}.availableStates",
-                    title: "Google Home Available States for ${sensorType.key}",
-                    type: "enum",
-                    multiple: "true",
-                    required: "true",
-                    options: GOOGLE_SENSOR_STATES[sensorType.key].descriptiveState,
-                )
-                input(
-                    name: "${deviceTrait.name}.sensorTypes.${sensorType.key}.descriptiveAttribute",
-                    title: "Hubitat Descriptive State Attribute for ${sensorType.key}",
-                    type: "text",
-                    required: "true",
-                    defaultValue: GOOGLE_SENSOR_STATES[sensorType.key].descriptiveAttribute
-                )
+                // if the sensor does not have a numeric state, do not allow the user to disable the descriptive state
+                if (GOOGLE_SENSOR_STATES[sensorType.key].numericUnits != "") {
+                    input(
+                        name: "${deviceTrait.name}.sensorTypes.${sensorType.key}.reportsDescriptiveState",
+                        title: "${sensorType.key} reports descriptive state",
+                        type: "bool",
+                        defaultValue: true,
+                        required: true,
+                        submitOnChange: true
+                    )
+                } else {
+                    deviceTrait.sensorTypes[sensorType.key].reportsDescriptiveState = true;
+                }
```

Comment:
This logic probably needs to go in the `traitFromSettings_SensorState` method.  It won't get persisted from here.

Reply:
Yes, I see what you are saying.  I was thinking it didn't matter, but I do need to pull it for constructing states/traits back to Google.

Reply:
Refactored and moved to the traitFromSettings_SensorState method.  I needed to change a couple checks to be value != false to deal with backwards compatibility (reportsDescriptiveState and reportsNumericState are both null until the sensor type is edited.